app/src/main/java/com/allan/boardbuddies/activities/EditNoteFragment.java

```diff
@@ -0,0 +1,59 @@
+package com.allan.boardbuddies.activities;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.inputmethod.InputMethodManager;
+import android.widget.EditText;
+
+import androidx.appcompat.widget.Toolbar;
+import androidx.fragment.app.Fragment;
+import androidx.lifecycle.ViewModelProvider;
+import androidx.navigation.Navigation;
+
+import com.allan.boardbuddies.R;
+import com.allan.boardbuddies.viewmodels.EditNoteViewModel;
+
+public class EditNoteFragment extends Fragment {
+    private EditText editTextTitle;
+    private EditText editTextContent;
+    private EditNoteViewModel editNoteViewModel;
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_edit_note, container, false);
+        Toolbar toolbar = view.findViewById(R.id.note_toolbar);
+
+        editNoteViewModel = new ViewModelProvider(requireActivity()).get(EditNoteViewModel.class);
+
+        toolbar.setNavigationOnClickListener(v -> {
+            if (editNoteViewModel.getLocalNote().getFileName() == null){ //if there is no local file: saveTextNote()
+                editNoteViewModel.saveNote(editTextTitle.getText().toString(), editTextContent.getText().toString());
+            } else if (!editNoteViewModel.getLocalNote().getTitle().equals(editTextTitle.getText().toString()) || !editNoteViewModel.getLocalNote().getContent().equals(editTextContent.getText().toString())) { //if local file has been changed
+                editNoteViewModel.deleteNote();
+                editNoteViewModel.saveNote(editTextTitle.getText().toString(), editTextContent.getText().toString());
+            }
+            System.out.println(Navigation.findNavController(requireActivity(), R.id.nav_host).popBackStack());
+        });
+
+        View extraView = view.findViewById(R.id.extra_scrollspace);
+        editTextTitle = view.findViewById(R.id.edit_text_note_title);
+        editTextContent = view.findViewById(R.id.edit_text_note_content);
+
+        editNoteViewModel.getSelectedPosition().observe(getViewLifecycleOwner(), selectedNotePosition -> {
```

Comment:
Still recommend wrapping all the necessary objects into a ViewState, and observing the whole thing.

The issue with the current approach is that you're assuming title and content could only change if selected position has changed - but imagine if you're using a network request instead of local data, what if the network request hasn't had time to complete yet?

Reply:
Don't need to do this if you just move selected position into a bundle argument

Reply:
now I'm not observing anything, it's just a bundle